nupic/algorithms/KNNClassifier.py

```diff
@@ -641,12 +641,23 @@ def infer(self, inputPattern, computeScores=True, overCategories=True,
       print "%s infer:" % (g_debugPrefix)
       print "  active inputs:",  _labeledInput(inputPattern,
                                                cellsPerCol=self.cellsPerCol)
-      print "  winner category:", winner
+      print "  winner categories:", winners
       print "  pct neighbors of each category:", inferenceResult
       print "  dist of each prototype:", dist
       print "  dist of each category:", categoryDist
 
-    result = (winner, inferenceResult, dist, categoryDist)
+    result = (winners, inferenceResult, dist, categoryDist)
+    numCategories = len(self._categoryList)
+
+    if numWinners > numCategories:
+          warnings.warn("Your requested number of categories exceeds the "
+                         "total number of categories learned. Returning %d "
+                         "categories..." %(numCategories))
+    if numWinners > self.k:
```

Comment:
This shouldn't make a difference. If `k` = 1, we still want a list of `numWinners` categories. The param `k` is used to calculate distances to potential classification categories. @mihail911 I can explain this further in person if you'd like.